File: Manali/ArraysAndHashing/LongestConsecutiveSequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums: return 0
        elif len(nums) == 1: return 1
        nums = set(nums)
        longest = 0
        for num in nums:
            if num-1 not in nums:
                length = 1
                while num+length in nums: length += 1
                longest = max(longest, length)
        return longest
        

TC: O(n)
SC: O(n)

File: parth/Week2/LongestConsecutiveSequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> uset(nums.begin(), nums.end());
        int longest = 0;
        for(int& n : nums)
        {
            if(uset.contains(n-1))
            {
                continue;
            }
            int currentLongest = 0;
            while(uset.contains(n))
            {
                currentLongest++;
                uset.erase(n);
                n++;
            }
            if(longest < currentLongest)
            {
                longest = currentLongest;
            }
        }
        return longest;
    }
};